File: getUmbrella.py
# Similar to coin change 
# Approach 1- DP (Bottom-Up)
def getUmbrella(requirements,sizes):
	dp=[requirements+1]*(requirements+1)
	dp[0]=0

	for size in sizes:
		for i in range(size,requirements+1):
			dp[i]=min(dp[i],dp[i-size]+1)

	return dp[requirements] if dp[requirements]!=requirements+1 else -1

print(getUmbrella(6,[3,5]))

File: getUmbrella.java
import java.util.*;
class Solution{
	public int getUmbrella(int requirements, int[] sizes){
		int max= requirements+1;
		int[] dp= new int [max];
		Arrays.fill(dp,max);
		dp[0]=0; 
		for (int i: sizes){
			for (int j=i; j<max;j++){
				dp[j]=Math.min(dp[j],dp[j-i]+1);
			}
		}
		return dp[requirements]>requirements ? -1:dp[requirements];
	}
}

class getUmbrella{
	public static void main(String[] args) {
		int requirements=6;
		int[] sizes={3,5};
		Solution s =new Solution();
		System.out.println(s.getUmbrella(requirements,sizes));
	}
}